extensions/amp-form/0.1/form-dirtiness.js

```diff
@@ -29,17 +30,24 @@ const SUPPORTED_TYPES = {
 export class FormDirtiness {
   /**
    * @param {!HTMLFormElement} form
+   * @param {!Window} win
    */
-  constructor(form) {
+  constructor(form, win) {
     /** @private @const {!HTMLFormElement} */
     this.form_ = form;
 
+    /** @private @const {!Window} */
+    this.win_ = win;
+
     /** @private {number} */
     this.dirtyFieldCount_ = 0;
 
     /** @private {!Object<string, boolean>} */
     this.isFieldNameDirty_ = map();
 
+    /** @private {!FormData} */
+    this.submittedFormData_ = this.takeFormDataSnapshot_();
```

Comment:
When constructing this class, do we have anything in the form? Does it really make sense of this first calculation or it's just making implementation easy?

Reply:
Yes this is to make the implementation easier. I could add a `null` check to the `hasFieldBeenSubmitted_` if that's preferable?

Reply:
Yeah let's make this null at the beginning.
It confuses me when first reading it, especially with the name "submitted" form data.